gnovm/pkg/gnolang/preprocess.go

```diff
@@ -882,17 +882,26 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 					// Left not const, Right not const ------------------
 					if n.Op == EQL || n.Op == NEQ {
 						// If == or !=, no conversions.
-					} else if lnt, ok := lt.(*NativeType); ok {
+					} else if ok, lnt := func() (bool, *NativeType) {
```

Comment:
Can you simplify this condition to something like this?
```go
} else if lnt, ok := lt.(*NativeType); ok && isNative(rt) {
```

Reply:
right it is better. Thx ^^

commit: https://github.com/gnolang/gno/pull/2016/commits/a8e4e7d3065205616836cc20913fef0278be6ba9